config/config_handler.go

```diff
@@ -37,40 +38,37 @@ func findConfigFile(path string, module string) string {
 		path += "/" + module + "/config.toml"
 		if fileExists(path) {
 			// file exists
-			return path
+			return path, nil
 		}
 	}
-	panic(errors.New("no config file found"))
+	return "", errors.New("no config file found")
 }
 
-// Save the Configuration in its current state
-// into the stored file path
-func Save(c *Configuration) {
+// Save the Configuration in its current state into the stored file path
+func Save(c *Configuration) error {
 	bytes, err := toml.Marshal(c)
 	if err != nil {
-		// TODO error handling
-		panic(err)
-	}
-	err = ioutil.WriteFile(c.path, bytes, 0644) //TODO what permissions for the file?
-	if err != nil {
-		// TODO error handling
-		panic(err)
+		return err
 	}
+	return ioutil.WriteFile(c.path, bytes, 0644)
 }
 
-// Looks for the configuration file and
-// returns a filled Configuration struct
-func New(path string, module string) *Configuration {
-	c := Configuration{}
-	c.path = findConfigFile(path, module)
+// Looks for the configuration file and returns a filled Configuration struct
+func New(path string, module string) (*Configuration, error) {
+	if p, err := findConfigFile(path, module); err != nil {
+		return nil, err
+	} else {
+		c := Configuration{}
+		c.path = p
 
-	// Read the config file
-	bytes, err := ioutil.ReadFile(c.path)
-	if err != nil {
-		panic(err)
-	}
+		bytes, err := ioutil.ReadFile(c.path)
+		if err != nil {
+			return nil, err
+		}
 
-	toml.Unmarshal(bytes, &c)
+		toml.Unmarshal(bytes, &c)
+		OverrideEnvConfiguration(&c)
```

Comment:
If we use this function here, we shall never save our configuration back to the file, because we would overwrite the configuration file with the environment variables in every case.

Reply:
Fair enough, any idea how to handle that case then?

Reply:
Mhh. Either we remember the original value, or we need some kind of "reload and over-overwrite" the environment variable changes.
Second approach sounds not reasonable.

Reply:
Or I remove it from New and make the call explicit then it’s the callers responsibility to take care about that.

Reply:
I am ok with that. The only case for `Save()`ing atm is the hostname, anyways. So we should probably check the hostname before the env variables ...(?)

Reply:
Isn’t hostname usually set as `HOSTNAME` env variable?

Reply:
yes. So you may just check the ENV var, instead of saving it in the config?